backend/src/services/userService.ts

```diff
@@ -90,6 +100,30 @@ const checkPassword = async (
     }
 };
 
+const changePassword = async (
+    userID: string,
+    currentPassword: string,
+    newPassword: string
+): Promise<RequestInfo> => {
+    const hash = await selectPasswordbyID(userID);
+    if (hash === null) return DEFAULT_ERROR;
+    try {
+        if (await bcrypt.compare(currentPassword, hash)) {
+            const passwordHash = await bcrypt.hash(newPassword, 10);
```

Comment:
I would suggest implementing a hashPassword(password: string): string mini function that would be used also in createUser function. This way modifying it in the future will be easy and it won't require finding all the places where this is done.